sympy/physics/control/lti.py

```diff
@@ -1422,3 +1441,520 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+        transposed_mat = self._expr_mat.transpose()
+        return _to_TFM(transposed_mat, self.var)
+
+    def elem_poles(self):
+        """
+        Returns the poles of each element of the ``TransferFunctionMatrix``.
+
+        .. note::
+            Actual poles of a MIMO system are NOT the poles of individual elements.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix
+        >>> tf_1 = TransferFunction(3, (s + 1), s)
+        >>> tf_2 = TransferFunction(s + 6, (s + 1)*(s + 2), s)
+        >>> tf_3 = TransferFunction(s + 3, s**2 + 3*s + 2, s)
+        >>> tf_4 = TransferFunction(s + 2, s**2 + 5*s - 10, s)
+        >>> tfm_1 = TransferFunctionMatrix([[tf_1, tf_2], [tf_3, tf_4]])
+        >>> tfm_1
+        TransferFunctionMatrix(((TransferFunction(3, s + 1, s), TransferFunction(s + 6, (s + 1)*(s + 2), s)), (TransferFunction(s + 3, s**2 + 3*s + 2, s), TransferFunction(s + 2, s**2 + 5*s - 10, s))))
+        >>> tfm_1.elem_poles()
+        [[[-1], [-2, -1]], [[-2, -1], [-5/2 + sqrt(65)/2, -sqrt(65)/2 - 5/2]]]
+
+        See Also
+        ========
+
+        elem_zeros
+
+        """
+        poles_list = []
+        for i in self.args[0]:
+            temp = []
+            for j in i:
+                if not isinstance(j, TransferFunction):
+                    j = j.doit()
+                temp.append(j.poles())
+            poles_list.append(temp)
+            temp = []
+        return poles_list
+
+    def elem_zeros(self):
+        """
+        Returns the zeros of each element of the ``TransferFunctionMatrix``.
+
+        .. note::
+            Actual zeros of a MIMO system are NOT the zeros of individual elements.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s
+        >>> from sympy.physics.control.lti import TransferFunction, TransferFunctionMatrix
+        >>> tf_1 = TransferFunction(3, (s + 1), s)
+        >>> tf_2 = TransferFunction(s + 6, (s + 1)*(s + 2), s)
+        >>> tf_3 = TransferFunction(s + 3, s**2 + 3*s + 2, s)
+        >>> tf_4 = TransferFunction(s**2 - 9*s + 20, s**2 + 5*s - 10, s)
+        >>> tfm_1 = TransferFunctionMatrix([[tf_1, tf_2], [tf_3, tf_4]])
+        >>> tfm_1
+        TransferFunctionMatrix(((TransferFunction(3, s + 1, s), TransferFunction(s + 6, (s + 1)*(s + 2), s)), (TransferFunction(s + 3, s**2 + 3*s + 2, s), TransferFunction(s**2 - 9*s + 20, s**2 + 5*s - 10, s))))
+        >>> tfm_1.elem_zeros()
+        [[[], [-6]], [[-3], [4, 5]]]
+
+        See Also
+        ========
+
+        elem_poles
+
+        """
+        zeros_list = []
+        for i in self.args[0]:
+            temp = []
+            for j in i:
+                if not isinstance(j, TransferFunction):
+                    j = j.doit()
+                temp.append(j.zeros())
+            zeros_list.append(temp)
+            temp = []
+        return zeros_list
```

Comment:
This is
```
return [element.doit().zeros() for row in self.args[0] for element in row]
```
It would probably be useful to add a `._flat()` method that returns all elements as a flattened list and then this would just be:
```
return [element.doit().zeros() for element in self._flat()]
```

Reply:
> This is
> 
> ```
> return [element.doit().zeros() for row in self.args[0] for element in row]
> ```

@oscarbenjamin, actually, the `zeros_list` variable mentioned above is not a flat list of zeros. I have anyways added the `_flat()` method as it would surely be useful in later phases.